src/Sylius/Bundle/OrderBundle/Remover/ExpiredCartsRemover.php

```diff
@@ -36,8 +37,13 @@ public function remove(): void
 
         $this->eventDispatcher->dispatch(new GenericEvent($expiredCarts), SyliusExpiredCartsEvents::PRE_REMOVE);
 
+        $interval = 0;
         foreach ($expiredCarts as $expiredCart) {
             $this->orderManager->remove($expiredCart);
+            if ($interval !== 0 && $interval % $this->batchSize === 0) {
+                $this->orderManager->flush();
+            }
+            $interval++;
```

Comment:
```suggestion
            $interval++;
            
            if ($interval % $this->batchSize === 0) {
                $this->orderManager->flush();
            }
```
or, to make it fancier 
```suggestion
            if (++$interval % $this->batchSize === 0) {
                $this->orderManager->flush();
            }
```

but probably even `$interval = $interval + 1;` to not overcomplicate will make the biggest sence